rio/subsystems/superstructure.py

```diff
@@ -0,0 +1,54 @@
+import wpilib
+
+from ntcore import NetworkTableInstance
+from commands2 import Subsystem
+from rev import CANSparkLowLevel, CANSparkMax, SparkMaxAbsoluteEncoder
+
+from constants import SuperStrucConstants
+
+
+class Superstructure(Subsystem):
+    def __init__(self) -> None:
+        super().__init__()
+
+        # dashboard
+        self.nt = NetworkTableInstance.getDefault()
+        self.sd = self.nt.getTable("SmartDashboard")
+
+        # motors
+        self.rotateMotor = CANSparkMax(
+            SuperStrucConstants.rotateID, CANSparkLowLevel.MotorType.kBrushless
+        )
+        self.flyMotor = CANSparkMax(
+            SuperStrucConstants.flyID, CANSparkLowLevel.MotorType.kBrushless
+        )
+
+        # encoders
+        self.flyEncoder = self.flyMotor.getEncoder()
+        self.rotateEncoder = self.rotateMotor.getAbsoluteEncoder(
+            SparkMaxAbsoluteEncoder.Type.kDutyCycle
+        )
+
+        # PID values
+        self.rotatePIDController = self.rotateMotor.getPIDController()
+
+        self.rotatePIDController.getP(SuperStrucConstants.kP)
+        self.rotatePIDController.getI(SuperStrucConstants.kI)
+        self.rotatePIDController.getD(SuperStrucConstants.kD)
+        self.rotatePIDController.getFF(SuperStrucConstants.kFF)
+
+    def periodic(self) -> None:
+        self.sd.putNumber("Thermals/rotate", self.rotateMotor.getMotorTemperature())
+        self.sd.putNumber("Thermals/fly", self.flyMotor.getMotorTemperature())
+
+    def fire(self, speed):
+        self.flyMotor.set(speed)
+
+    def gotoPOS(self, angle: float):
+        self.rotatePIDController.setReference(angle, CANSparkMax.ControlType.kPosition)
+
+    def rotateCurrentLimit(self, current):
```

Comment:
Maybe call these with some startup-defaults in `__init__`?

Reply:
do you mean the current limit commands or the set reference? I'm pretty sure which one you mean but if you could clarify in case I'm forgetting some specific instance.